### PR TITLE
Feat: Logging

### PR DESCRIPTION
**Pull request by tts-jwiebe**
Added Logging, it creates a 'log.txt'. #25 
As discussed in the Issue i decided to go with the file. 

img

I've changed the time to be in 24H Format with leading zeros. As just using `toLocaleTimeString` could lead to different outputs.

### CODE REVIEW

**Comment by tts-jwiebe** [comment]
I would suggest to extend the tests with a POST Request, which should post some data.
In my tests i've used ```curl --header "Content-Type: application/json" --request POST --data '{"username":"xyz","password":"xyz"}' http://localhost:8080/index.html```

File: server.js [hunk 1]
```diff
@@ -35,7 +35,24 @@ function deploy(config, ready) {
         return path.posix.join(...url.split(path.sep));
     }
 
+    let logger = fs.createWriteStream('log.txt', {
+        flags: 'a'
+    });
+    let writeLine = (line) => logger.write(`\n${line}`);
+
     server.on('request', (request, response) => {
+        let now = new Date().toLocaleTimeString('en-US', {  hour: "2-digit", minute: "2-digit", hour12: false })
```
Comment (Eun) [inline server.js:44]: I would prefer defining the format beforehand (in the `deploy` function) using
```javascript
const formatTime = Intl.DateTimeFormat('en-US', { hour: "2-digit", minute: "2-digit", second:"2-digit", hour12: false }).format
```

And using the `formatTime(Date.now())` in the `request` function

**Review by Eun — CHANGES_REQUESTED** [review]
Thanks for your contribution!

I have 3 requests:
1. Could you make the log functionality optional? It should be possible to enable/disable the functionality.
2. Could you add a json log format?
3. Could you make the logfile name configurable? Currently the file will be created in the git workdirectory itself leaving the git workspace with unstaged changes. 

1, 2 & 3 could be combined by having an option like:
```
log: off
log: log.json
log: log.txt
```

You also mentioned adding a test. I would also like to see a test in this PR here for the new functionality.

**Comment by tts-jwiebe** [comment]
JSON seems to be an inappropiate format for logging. 
Because in the worst case i would need to read the whole file which might be >= 100 MB. Parse it's JSON content and finally append the single JSON log object just to write the whole potential big file back. 

I just can't blindly append which i do with text logging as this would lead to wrong json.

File: server.js [hunk 2]
```diff
@@ -34,14 +38,24 @@ function deploy(config, ready) {
...
-    });
-    let writeLine = (line) => logger.write(`\n${line}`);
+    
+    let writeLine = (line) => {
+        if (config.log != "off") {
+            let path = config.log;
+            // ensure .txt file extension is available
+            if (!path.endsWith(".txt")) {
+                path += ".txt";
+            }
```
Comment (Eun) [inline server.js:48]:
```suggestion
```
this is not needed. If the developer wants to create a file without extension, so be it

File: server.js [hunk 3]
```diff
@@ -34,14 +38,24 @@ function deploy(config, ready) {
...
         return path.posix.join(...url.split(path.sep));
     }
-
-    let logger = fs.createWriteStream('log.txt', {
-        flags: 'a'
-    });
-    let writeLine = (line) => logger.write(`\n${line}`);
+    
+    let writeLine = (line) => {
+        if (config.log != "off") {
```
Comment (Eun) [inline server.js:43]:
```suggestion
        if (config.log !== "off") {
```

**Review by Eun — CHANGES_REQUESTED** [review]
You are correct about the json format. 2 minor things and then we ar egood to merge :)

**Review by Eun — APPROVED** [review]

**Comment by Eun** [comment]
Thanks!

**Pull request merged**